docs/doxygen/logging/logging.dox

```diff
@@ -0,0 +1,45 @@
+/**
+@page logging Logging Infrastructure
+@brief This page contains an overview of the logging infrastructure used by the C-SDK Libraries.
+
+<p>
+The C-SDK libraries utilize one of the following 4 macro interfaces for logging within the library for different verbosity levels:
+- `LogError` for **Error** level
+- `LogWarn` for **Info** level
+- `LogInfo` for **Info** level
+- `LogDebug` for **Debug** level
+
+**Note: By default, each library contains empty definitions of the above logging macros, thereby, generating no code for logging calls.**
+
+To enable logging within the libraries, the customer application should provide implementation of these macros through the library-specific config file (for example, the `core_mqtt_config.h` file for MQTT library).
+Implementation of these macros can be provided as mapping to the customer's platform-specific logging implementation.
+</p>
+
+@section design Design Architecture
+@brief The logging infrastructure architecture follows a dependency inversion design principle of allowing the customer application to inject implementation of logging interface macros through the library-specific config file (like `core_mqtt_config.h`).
+
+@image html logging_architecture.png width=100%
+
+@section reference_implementation Reference Implementation
+@brief For reference example of a POSIX-specific logging implementation, refer to the [demos/logging-stack](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master/demos/logging-stack) folder from the root directory of the C-SDK repository.
+
+The [logging_stack.h](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/master/demos/logging-stack/logging_stack.h) contains the definition of all the logging interface macros (`LogError`, `LogWarn`, `LogInfo`, `LogDebug`) that eventually map to the standard C function, `printf`.
```

Comment:
I feel like you could just add this file to doxygen instead of having to link to the GitHub repo

Reply:
Sure, that's a good suggestion